pkg/deployments/manager.go

```diff
@@ -39,6 +40,10 @@ type Manager struct {
 
 	ScaleDownPeriod time.Duration
 
+	deploymentsMtx sync.RWMutex
+	// deployments is a set of deployment names that are managed by the Manager
+	deployments map[string]struct{}
```

Comment:
This appears to be an grow-only map as implemented. Why do we need to track this here?

Reply:
so we can do HasModel because stats aggregates stats of all deployments and autoscaler is active on all deployments. I had to do this to get it to work.